### PR TITLE
Refactor main page to improve performance.

### PR DESCRIPTION
**Pull request by MagneFire**
The main settings page has been laggy for a while now.
The solution is to replace the `Flickable` with a `ListView`. Which is exactly what this PR does.
In addition the page has been redesigned so that it feels and looks better.
Thanks for @eLtMosen for helping with the design of this new settings page.

https://user-images.githubusercontent.com/7857908/149674103-5a96d060-1cf2-4270-a45b-fdbcf0f3e778.mp4

This depends on https://github.com/AsteroidOS/qml-asteroid/pull/17.

### CODE REVIEW

**Comment by FlorentRevest** [comment]
This is overall a much needed performance improvement and thanks for that.

However I'm not so happy with the current look of things :/ I dislike that the icon overlays text, this makes the UI feel very confused, there is space for everything but somehow we show things on top of each others. Have you considered at least making the ListView smaller (starting from the bottom of the icon up until the bottom of the screen) ? Then text will naturally fade and disappear before reaching the Icon. That also means that the selected item will no longer be in the center of the screen. I guess one could also reduce the size of the ListView from both ends but then this will look very small. One could apply a strong transparent background on the upper part than on the lower part but this would probably look weird.

I would honestly prefer if the icon were on the left side of the highlighted text. I understand this gives less space to long strings on some translations but we could optimize the default case (english), cap the dimension of the Label and if the string gets longer than that why not animate it a bit like what we do in the Music app. Idk I feel the UI would feel a lot more natural and the patterns would be more easily recognizable.

**Comment by MagneFire** [comment]
Thanks for the feedback! And big thanks to being honest about it. We've been looking at this for quite some time so it's good to get a fresh look on this :smile:.

We did indeed considered reducing the size of the `ListView` but this would, like you said, reduce the overall amount of items that you can view at a time, in addition it would introduce a void at the bottom of the page (A sort of in-balance in the UI style?).

@eLtMosen and I have adjusted the design to incorporate the 'notch' that was also introduced in the calculator app, essentially the 'strong transparent background' you mentioned. This should mitigate some/all of the issues.
 
https://user-images.githubusercontent.com/7857908/149680390-0e7f47bc-a3e5-43eb-8809-e67534980025.mp4

**Comment by eLtMosen** [comment]
> This is overall a much needed performance improvement and thanks for that.
> 
> However I'm not so happy with the current look of things :/ 

Looks like we have a classic case of three quarter finished work shown too early :D
Now it is kind of much to explain, but indeed you are right with your observations.
The Text in the video is looking messy when floating below the text. But that was a minor issue for long time and we optimised on other ends, hoping for a solution for that problem to come around at some point.
I am looking into this page since weeks now and proposed the tumbler to magnefire, like you noted to move the icon out of the way of long strings. And generally be able to make the font larger. After many trials the icon top position was chosen as only valid position to keep an icon at all. We kept that in all further trials since placing it anywhere else felt imbalanced. Or the icon was not really noticable. Long story short the last open issue was how to mitigate the text/icon overlap by either adjusting the spinner. Or finding a proper way to frame the icon.
Which you did this evening by suggesting to play on the notch themed calculator a little more.
Imo thats a great idea to raise consistency by creating yet another consistent element over many pages.
We had that notch in calculator all along where i copied it from for the calculator.
Long story short, @magnefire did the PR just some hours too early to present the very much improved state that developed just during the last 2 hours.
I think he will post a video and update the PR soon for you to see what i refer to :D

**Comment by FlorentRevest** [comment]
Mhh well the "notch" does look a bit better but still I have issues with this whole UI pattern. This feels a bit artificial to just place an icon *somewhere* "because we have to".
If having an icon is so problematic, maybe we should just drop it. I don't know.

Also, after having discussed with MagneFire, it looks like the reason why we're not using a listview in the style of the current alternative app launcher is because performance of apps is bad and, interestingly, noticeably worse than asteroid-launcher itself. First of all I think this is an issue to investigate, the performance of apps should really be the same as the performance of the launcher and this is certainly something to look into. And in an ideal world UI/UX should not be constrained by this kind of performance limitations.

So I'm ambivalent on this. I propose we:
- Try to find the reason why apps are slow in the first place.
  - If we find a solution, consider a listview in the style that MagneFire had experimented with at first
  - If we don't find a solution, try to consider what it would look without any icons. I don't like this but maybe this would feel more natural than just putting an icon for the sake of it
    - If this does not look good or is too big of a UX regression, merge this as a last resort. It's not *that* bad but this feels like a weird pattern to show an icon somewhere else than where the action is happening.

**Comment by MagneFire** [comment]
Closing this as we agreed that this isn't the right approach. Leaving the design as is.
https://github.com/AsteroidOS/meta-smartwatch/pull/64 seems to greatly improve the performance already.
